pkg/ddc/alluxio/transform_volumes.go

```diff
@@ -54,6 +57,38 @@ func (e *AlluxioEngine) transformMasterVolumes(runtime *datav1alpha1.AlluxioRunt
 	return err
 }
 
+func (e *AlluxioEngine) transformEncryptOptionsToMasterVolumes(dataset *datav1alpha1.Dataset, value *Alluxio) (options map[string]string) {
+	for _, m := range dataset.Spec.Mounts {
+		if common.IsFluidNativeScheme(m.MountPoint) {
+			continue
+		}
+		options = make(map[string]string)
+		for _, encryptOpt := range append(dataset.Spec.SharedEncryptOptions, m.EncryptOptions...) {
+			secretName := encryptOpt.ValueFrom.SecretKeyRef.Name
```

Comment:
I am wondering if  a new key is added,  there will be a new file for this key?

Reply:
Yes, see https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod.
However, should we check the secret key file exist before executing `alluxio fs mount` in script "mount.sh"?